### PR TITLE
chore: Fix build and test timeouts

### PR DESCRIPTION
**Pull request by sauravdas1997**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 1]
```diff
@@ -723,6 +725,9 @@ class _InsightsHandler {
         }
 
         const req = this._requestQueueHandler.add(uploadData)
+        if (this._teardownInvoked && req.data) {
+            await uploadEventData(req.data, req.url)
+        }
         if (req.proceed && req.data) {
             await uploadEventData(req.data, req.url)
         }
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/insights-handler.ts:733]:
```suggestion
        if (req.data && (this._teardownInvoked || req.proceed)) {
            await uploadEventData(req.data, req.url)
        }
```

File: packages/wdio-browserstack-service/src/util.ts [hunk 2]
```diff
@@ -1159,3 +1161,15 @@ export async function uploadLogs(user: string | undefined, key: string | undefin
 
     return response
 }
+
+export function setupExitHandlers() {
+    const __filename = fileURLToPath(import.meta.url)
+    const __dirname = path.dirname(__filename)
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/util.ts:1167]: These should be defined at the top of the file.

File: packages/wdio-browserstack-service/src/cleanup.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        }
+    }
+    static async executeObservabilityCleanup() {
+        BStackLogger.debug('Executing observability cleanup')
+        if (process.env.IS_TESTOPS_SESSION?.toLowerCase() === 'true') {
+            await stopBuildUpstream()
+            if (process.env.BS_TESTOPS_BUILD_HASHED_ID) {
+                BStackLogger.info(`\nVisit https://observability.browserstack.com/builds/${process.env.BS_TESTOPS_BUILD_HASHED_ID} to view build report, insights, and many more debugging information all at one place!\n`)
+            }
+        }
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/cleanup.ts:22]: return early:

```suggestion
        if (process.env.IS_TESTOPS_SESSION?.toLowerCase() !== 'true') {
            return
        }
        
        await stopBuildUpstream()
        if (process.env.BS_TESTOPS_BUILD_HASHED_ID) {
            BStackLogger.info(`\nVisit https://observability.browserstack.com/builds/${process.env.BS_TESTOPS_BUILD_HASHED_ID} to view build report, insights, and many more debugging information all at one place!\n`)
        }
```

File: packages/wdio-browserstack-service/src/cleanup.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export default class BStackCleanup {
+    static async startCleanup() {
+        try {
+            if (process.env.TESTOPS_BUILD_STOPPED?.toLowerCase() !== 'true') {
+                await this.executeObservabilityCleanup()
+            }
+        } catch (err) {
+            const error = err as string
+            BStackLogger.error(error)
+        }
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/cleanup.ts:13]: same

```suggestion
        if (process.env.TESTOPS_BUILD_STOPPED?.toLowerCase() !== 'true') {
            return
        }
        try {
            await this.executeObservabilityCleanup()
        } catch (err) {
            const error = err as string
            BStackLogger.error(error)
        }
```

**Comment by christian-bromann** [comment]
@sauravdas1997 mind signing the CLA?

**Comment by sauravdas1997** [comment]
Hey @christian-bromann Signing done, All comments resolved. Thanks

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Pull request merged**